### PR TITLE
Change `AssetHierarchy` validation: Assume parent ID linked assets are valid

### LINKED ISSUE #1282: partial hierarchy with asset linked to existing asset by parent_id
**System information (please complete the following information):**
 - OS: Ubuntu
 - Python Version: python3.10
 - SDK Version: 6.6.0

**Describe the bug**
According to the documentation it is allowed to create “partial hierarchies” ([docs](https://cognite-sdk-python.readthedocs-hosted.com/en/latest/core_data_model.html#create-asset-hierarchy)), i.e which depend on existing assets.

It is said explicitly, that "asset with specified parent_id are considered valid" which is not true in the current version of sdk

**To Reproduce**
Runnable code reproducing the error.
```
from cognite.client.data_classes import (
    Asset,
    AssetHierarchy,
)

id_of_potentially_existing_asset = 123
asset = Asset(parent_id=id_of_potentially_existing_asset, name="new asset", external_id="test.external.id")
hierarchy = AssetHierarchy([asset])
print(hierarchy.is_valid())  # returns False
```
**Expected behavior**
should pass validation

**Additional context**

The piece of code below doesn’t take into account an option when “parent_id was provided (and is from existing asset, so not part of the partial hierarchy) and parent_external_id was not provided”

https://github.com/cognitedata/cognite-sdk-python/blame/449109b0325d0f180983a31e7ecb6e6ae97784c7/cognite/client/data_classes/assets.py#L681-L701

```
            if (pxid := asset.parent_external_id) is (pid := asset.parent_id) is None:
                roots.append(asset)
            elif pxid is not None and pid is not None:
                unsure_parents.append(asset)  # Both parent links are given
            elif pxid not in xid_count:
                orphans.append(asset)  # Only parent XID given, but not part of assets given
```

### PR DESCRIPTION
**Pull request by haakonvt**
## Description
Resolves #1282 reported by @vindex10: Assets only linking a parent through ID still end up as part of "orphan assets" instead of being "assumed valid" as the documentation says (and the rest of the code...).

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: tests/tests_unit/test_data_classes/test_assets.py [hunk 1]
```diff
@@ -256,6 +257,22 @@ def test_validate_asset_hierarchy__orphans_given_ignore_false(self, exc_type):
         with pytest.raises(exc_type, match=r"Issue\(s\): 1 orphans$"):
             ahv.is_valid(on_error="raise")
 
+    @pytest.mark.parametrize("n", [1, 2, 3, 10])
+    def test_validate_asset_hierarchy__orphans_given_ignore_false__all_parent_external_id(self, n):
+        assets = [Asset(name=c, external_id=c, parent_external_id="foo") for c in string.ascii_letters[:n]]
+        ahv = AssetHierarchy(assets, ignore_orphans=False).validate(on_error="ignore")
```
Comment (doctrino) [inline tests/tests_unit/test_data_classes/test_assets.py:263]: What is `ahv` can you write it out? Difficult to interpret this test now.

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,11 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [6.8.1] - 2023-07-08
+### Changed
+- The `AssetHierarchy` class would consider assets linking its parent only by ID as orphans, contradicting the
+  docstring stating "All assets linking a parent by ID are assumed valid". This is no longer the case.
```
Comment (haakonvt) [inline CHANGELOG.md:23]:
```suggestion
- The `AssetHierarchy` class would consider assets linking their parent only by ID as orphans, contradicting the
  docstring stating "All assets linking a parent by ID are assumed valid". This is now true (they are no longer
  considered orphans).
```

File: tests/tests_unit/test_data_classes/test_assets.py [hunk 3]
```diff
@@ -256,6 +257,22 @@ def test_validate_asset_hierarchy__orphans_given_ignore_false(self, exc_type):
...
+        assert len(assets) == len(ahv.orphans) == n
+        with pytest.raises(CogniteAssetHierarchyError, match=rf"Issue\(s\): {n} orphans$"):
+            ahv.is_valid(on_error="raise")
+
+    @pytest.mark.parametrize("n", [0, 1, 2, 3, 10])
+    def test_validate_asset_hierarchy__orphans_given_ignore_false__all_parent_id(self, n):
+        assets = [Asset(name=c, external_id=c, parent_id=ord(c)) for c in string.ascii_letters[:n]]
+        ahv = AssetHierarchy(assets, ignore_orphans=False).validate(on_error="ignore")
```
Comment (doctrino) [inline tests/tests_unit/test_data_classes/test_assets.py:271]: Same here, what is `ahv`?

**Review by doctrino — APPROVED** [review]
Looks good to me, only things is to avoid the abbreviation in the test which makes it hard to read.

File: tests/tests_unit/test_data_classes/test_assets.py (hunk 1 above)
Comment (doctrino) [reply tests/tests_unit/test_data_classes/test_assets.py:263]: Does it mean `valid_asset_hierarchy`?

Comment (haakonvt) [reply tests/tests_unit/test_data_classes/test_assets.py]: Class used to be named `AssetHierarchyValidator`. Changed to hierarchy which is short and unmistakable 😅

**Pull request merged**